overlord/snapstate/refresh.go

```diff
@@ -184,3 +157,87 @@ func (err *BusySnapError) Error() string {
 func (err BusySnapError) Pids() []int {
 	return err.pids
 }
+
+// doHardRefreshCheck performs the complete hard refresh check interaction.
```

Comment:
this is now run inhibiting the snap so just "Check" is probably not a good name for this anymore